File: C/242.valid-anagram.c
/*
 * @lc app=leetcode id=242 lang=c
 *
 * [242] Valid Anagram
 */

// @lc code=start
bool isAnagram(char *s, char *t)
{
  int count[26] = {0};

  for (int i = 0; s[i] != '\0'; i++)
  {
    count[s[i] - 'a']++;
  }

  for (int i = 0; t[i] != '\0'; i++)
  {
    count[t[i] - 'a']--;
  }

  for (int i = 0; i < 26; i++)
  {
    if (count[i] != 0)
    {
      return false;
    }
  }

  return true;
}
// @lc code=end

File: Rust/242.valid-anagram.rs
/*
 * @lc app=leetcode id=242 lang=rust
 *
 * [242] Valid Anagram
 */

// @lc code=start
impl Solution {
    pub fn is_anagram(s: String, t  : String) -> bool {
        let mut count = [0; 26];

        for c in s.chars() {
            count[(c as usize) - ('a' as usize)] += 1;
        }

        for c in t.chars() {
            count[(c as usize) - ('a' as usize)] -= 1;
        }

        for &c in count.iter() {
            if c != 0 {
                return false;
            }
        }

        true
    }
}
// @lc code=end
